pandas/tests/dtypes/test_dtypes.py

```diff
@@ -991,3 +991,8 @@ def test_is_dtype_no_warning(check):
 
     with tm.assert_produces_warning(None):
         check(data["A"])
+
+
+def test_check_dtype_empty_column():
+    data = pd.DataFrame({"a": [1, 2]}, columns=["b"], dtype=int)
+    assert data.b.dtype is np.dtype("int")
```

Comment:
maybe `==` instead of `is`?

Reply:
Isn't `np.dtype("int")` a singleton class such that `is`is more appropriate here?

Reply:
but `data.b.type` is also it, right? would preferably use `is` for identity checks only.

have no strong opinion anyway. thanks for the pr!

Reply:
Thanks for the feedback, I will change it.